core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -792,13 +792,22 @@ class PartKeyLuceneIndex(ref: DatasetRef,
     logger.info(s"Refreshed index searchers to make reads consistent for dataset=$ref shard=$shardNum")
   }
 
+
+  /**
+   * Remove leading anchor &#94; and ending anchor $.
+   * @param regex the orignal regex string.
+   * @return the regex string without anchors.
+   */
+  private def removeRegexAnchors(regex : String) : String = {
+    regex.stripPrefix("^").stripSuffix("$")
```

Comment:
Should we be careful about escape characters? I forget the exact methods, but maybe something like:
```
regex.stripPrefix("^")
// match (possibly zero) escaped backslashes, then a final "$" anchor; remove the anchor
.strip(matchRegex="(\\\\)*(\$)", replaceSecondGroupWith="")
```

(Let me know if I'm overthinking things and there's some reason we shouldn't do this.)

Reply:
Thank you. 
fixed the code. So it can handle case like """$"""  """\$"""  """\\$"""
The corresponding regex without anchor $ are """"""    """\$"""  """\\"""

Reply:
tested against Prometheus. '\$' is a dollar sign, whereas '$' is not.